app/services/unpublish_service.rb

```diff
@@ -2,16 +2,10 @@
 
 class UnpublishService
   def withdraw(edition, public_explanation, user = nil)
-    Document.transaction do
+    Document.transaction(requires_new: true) do
```

Comment:
Great spot - that was an interesting read! Hopefully we can refactor our controllers/services so that we're less likely to get caught-out by this sort of thing - it seems pretty niche.